extmod/uasyncio/funcs.py

```diff
@@ -10,40 +10,41 @@ async def wait_for(aw, timeout, sleep=core.sleep):
         return await aw
 
     def runner(waiter, aw):
-        nonlocal status, result
         try:
             result = await aw
-            s = True
+            status = True
         except BaseException as er:
-            s = er
-        if status is None:
-            # The waiter is still waiting, set status for it and cancel it.
-            status = s
-            waiter.cancel()
+            result = None
+            status = er
+        if waiter.data is None:
```

Comment:
Wouldn't it be better to check that it is not instance of core.CancelledError?

Reply:
I tried this, some tests will fail.  Eg when `waiter` is waiting on `runner` (in that case `waiter.data` will point to the `runner` task).